src/settings/settingsExport.ts

```diff
@@ -0,0 +1,114 @@
+import GioUnix from 'gi://Gio';
+import GLib from 'gi://GLib';
+import Settings from '@settings/settings';
+
+const dconfPath = '/org/gnome/shell/extensions/tilingshell/';
+const excludedKeys = [
+    Settings.SETTING_LAYOUTS_JSON,
+    Settings.SETTING_LAST_VERSION_NAME_INSTALLED,
+    Settings.SETTING_OVERRIDDEN_SETTINGS,
+];
+
+export default class SettingsExport {
+    dconfDump: string;
+
+    constructor() {
+        this.dconfDump = this.dumpDconf();
+        this.excludeKeys();
+    }
+
+    static importFrom(content: string) {
+        this.restoreToDefault();
+
+        // Flush overridden settings back into system
+        Settings.set_active_screen_edges(false);
+        Settings.set_enable_move_keybindings(false);
+
+        const proc = GioUnix.Subprocess.new(
+            ['dconf', 'load', dconfPath],
+            GioUnix.SubprocessFlags.STDIN_PIPE,
+        );
+
+        const stdin = proc.get_stdin_pipe();
+        if (stdin) {
+            stdin.write(content, null);
+            stdin.close(null);
+            proc.wait(null);
+        }
+        if (!proc.get_successful()) {
+            throw new Error(
+                'Failed to import dconf dump file. Restoring to default...',
+            );
+        }
+    }
+
+    static restoreToDefault() {
+        const procList = GioUnix.Subprocess.new(
+            ['dconf', 'list', dconfPath],
+            GioUnix.SubprocessFlags.STDOUT_PIPE |
+                GioUnix.SubprocessFlags.STDERR_PIPE,
+        );
+
+        const [, list] = procList.communicate_utf8(null, null);
+        procList.wait(null);
+
+        if (!procList.get_successful())
+            throw new Error('Failed to retrieve dconf keys');
+
+        const keys = list
+            .split('\n')
+            .filter((key) => key.length > 0 && !excludedKeys.includes(key));
+
+        console.log(keys);
+
+        for (const key of keys) {
+            const procReset = GioUnix.Subprocess.new(
+                ['dconf', 'reset', '-f', dconfPath + key],
```

Comment:
This can be done without a subprocess: `giosettings.reset(key_name);`, where `giosettings` is an object of the class `Gio.Setting`